tests/test_django.py

```diff
@@ -684,8 +684,8 @@ def test_with_content(self):
         self.assertEqual(13, o.animage.height)
         self.assertEqual('django/example.jpg', o.animage.name)
 
-        i = Image.open(os.path.join(settings.MEDIA_ROOT, o.animage.name))
-        colors = i.getcolors()
+        with Image.open(os.path.join(settings.MEDIA_ROOT, o.animage.name)) as i:
+            colors = i.getcolors()
         # 169 pixels with rgb(254, 0, 0)
         self.assertEqual([(169, (254, 0, 0))], colors)
         self.assertEqual('JPEG', i.format)
```

Comment:
`i` is used here after the resource is closed. Does not appear to cause a test failure, but I suggest we assume a block scope for the `i` variable here to avoid confusion. Same with other cases.

Reply:
`format` [is an attribute of the `Image` class](https://github.com/python-pillow/Pillow/blob/65f3e44b3d7156cdf81b5d78abecec4fe8d9ed5a/src/PIL/Image.py#L534) and [does not require to read from disk, it’s either passed when constructing the image or guessed from file extension](https://github.com/python-pillow/Pillow/blob/65f3e44b3d7156cdf81b5d78abecec4fe8d9ed5a/src/PIL/Image.py#L1948-L1951). I left it out of the context manager, because it does not require to access the disk.

`getcolors()` requires to read from disk (it calls [`load()`](https://github.com/python-pillow/Pillow/blob/65f3e44b3d7156cdf81b5d78abecec4fe8d9ed5a/src/PIL/Image.py#L1247)).

Reusing context managers variables outside of the block is an established practice, for example [`unittest.assertRaises`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises) recommends keeping a reference to the context manager. It performs assertions after the block and does not keep resources when they aren’t needed.

Reply:
Gotcha, I think we are fine merging then.